solana/solana-ibc/programs/solana-ibc/src/transfer/mod.rs

```diff
@@ -237,6 +237,18 @@ impl ibc::Module for IbcStorage<'_, '_> {
                 acknowledgement,
                 relayer,
             );
+
+        // refund fee if there was an error on the counterparty chain
+        if result.1.is_err() {
+            let store = self.borrow();
+            let accounts = &store.accounts;
+            let receiver = accounts.receiver.clone().unwrap();
+            let fee_collector = accounts.fee_collector.clone().unwrap();
+            **fee_collector.try_borrow_mut_lamports().unwrap() -=
+                crate::FEE_AMOUNT_IN_LAMPORTS;
```

Comment:
Perhaps introduce a new constant `REFUND_FEE_AMOUNT_IN_LAMPORTS` which for now would equal `FEE_AMOUNT_IN_LAMPORTS`? Potentially we might want to refund less than fee since even if things failed we bared some costs.

Do we actually control timeouts? Or can user set very short timeout and effectively force a failure?

Reply:
a user can change the timeout and force the failure

Reply:
i think it makes sense to use `REFUND_FEE_AMOUNT_IN_LAMPORTS`